scripts/quests/sandoria/PLD_AF1_Sharpening_The_Sword.lua

```diff
@@ -0,0 +1,160 @@
+                player:getMainLvl() >= xi.settings.main.AF1_QUEST_LEVEL and
+                player:getQuestStatus(xi.quest.log_id.SANDORIA, xi.quest.id.sandoria.FATHER_AND_SON) == QUEST_COMPLETED and
+                player:getCharVar("Quest[0][4]Prog") == 0 -- must have turned the willow fishing rod back
+        end,
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Ailbeche'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:getCharVar("Quest[0][90]Prog") == 0 then
+                        return quest:progressEvent(45) -- Starts Quest.
+                    elseif player:getCharVar("Quest[0][90]Prog") == 1 then --declined first
+                        return quest:event(43)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [45] = function(player, csid, option, npc)
+                    if option == 1 then -- Accept quest.
+                        quest:begin(player)
+                        quest:setVar(player, 'Prog', 2)
+                    else
+                        quest:setVar(player, 'Prog', 1) -- Declined First offering.
+                    end
+                end,
+
+                [43] = function(player, csid, option, npc)
+                    if option == 1 then -- Accept quest.
+                        quest:begin(player)
+                        quest:setVar(player, 'Prog', 2)
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Section: Quest accepted
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Ailbeche'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:hasKeyItem(xi.ki.ORDELLE_WHETSTONE) then
+                        return quest:progressEvent(44)
+                    else
+                        return quest:event(42)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [44] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:delKeyItem(xi.ki.ORDELLE_WHETSTONE)
+                    end
+                end,
+            },
+
+        },
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Sobane'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') >= 2 then
+                        return quest:progressEvent(52) -- Finish quest.
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [52] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                end,
+            },
```

Comment:
The comment says this finishes quest? When it seems the block above in event 44 is what completes the quest.
Moreover, this block will loop infinitely. The player will always be >=2 when this event always sets it to 3. Creating an infinite loop in this dialogue.

Reply:
oof I copied the line from above and didnt delete the comment O.o